src/js/components/AppStatusComponent.jsx

```diff
@@ -29,6 +30,10 @@ var AppStatusComponent = React.createClass({
     showSummary: React.PropTypes.bool
   },
 
+  shouldComponentUpdate: function (nextProps) {
```

Comment:
This shouldn't be neccessary, I think React takes care of that itself.
Did you proved if this has any effect by console.log() in the render function?